synapse/util/caches/expiringcache.py

```diff
@@ -114,11 +119,14 @@ def _prune_cache(self):
 
         logger.debug(
             "[%s] _prune_cache before: %d, after len: %d",
-            self._cache_name, begin_length, len(self._cache)
+            self._cache_name, begin_length, len(self)
         )
 
     def __len__(self):
-        return len(self._cache)
+        if self.iterable:
+            return sum(len(value.value) for value in self._cache.itervalues())
```

Comment:
This is probably sub-optimal since it iterates the entire cache. You will probably need to store the current size somewhere and update it as things are added or removed.